lib/archethic/mining/validation_context.ex

```diff
@@ -1261,4 +1261,28 @@ defmodule Archethic.Mining.ValidationContext do
     |> Enum.map(&Enum.at(storage_nodes, &1))
     |> Enum.reject(&Utils.key_in_node_list?(chain_storage_nodes, &1.first_public_key))
   end
+
+  @doc """
+  Get the first available error or nil
+  """
+  @spec get_first_error(t()) :: atom()
+  def get_first_error(ctx = %__MODULE__{}) do
```

Comment:
An other way to write this function using pattern matching is : 
```elixir
def get_first_error(%__MODULE__{
      validation_stamp: %ValidationStamp{error: nil},
      cross_validation_stamps: cross_validation_stamps
    }) do
  cross_validation_stamps
  |> Enum.reduce_while(nil, fn
    %CrossValidationStamp{inconsistencies: []}, nil -> {:cont, nil}
    %CrossValidationStamp{inconsistencies: [first_error | _rest]}, nil -> {:halt, first_error}
  end)
end

def get_first_error(%__MODULE__{validation_stamp: %ValidationStamp{error: error}}), do: error
```

It may be more consistent with the rest of the project as a lot of function use pattern matching in priority.
As you prefer